File: app/src/main/java/com/kacper/krakgo/model/Place.java
package com.kacper.krakgo.model;

import android.os.Parcel;
import android.os.Parcelable;

/**
 * Created by kacper on 10/03/2018.
 */

public class Place implements Parcelable{
    private Double latitude;
    private Double longitude;
    private String display_name;
    private String photo_url;
    private String reviewsID;

    public Place() {
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel parcel, int i) {
        parcel.writeDoubleArray(new double[]{
                latitude,
                longitude
        });
        parcel.writeStringArray(new String[]{
                display_name,
                photo_url,
                reviewsID
        });
    }
    public Place(Parcel in){
        String[] strings = new String[3];
        double[] doubles = new double[2];

        in.readDoubleArray(doubles);
        in.readStringArray(strings);

        this.display_name = strings[0];
        this.photo_url = strings[1];
        this.reviewsID = strings[2];

        this.latitude = doubles[0];
        this.longitude = doubles[1];
    }
    public static final Parcelable.Creator CREATOR = new Parcelable.Creator() {
        public Place createFromParcel(Parcel in) {
            return new Place(in);
        }

        public Place[] newArray(int size) {
            return new Place[size];
        }
    };

    public Place(Double latitude, Double longitude, String display_name, String photo_url, String reviewsID) {
        this.latitude = latitude;
        this.longitude = longitude;
        this.display_name = display_name;
        this.photo_url = photo_url;
        this.reviewsID = reviewsID;
    }

    public Double getLatitude() {
        return latitude;
    }

    public void setLatitude(Double latitude) {
        this.latitude = latitude;
    }

    public Double getLongitude() {
        return longitude;
    }

    public void setLongitude(Double longitude) {
        this.longitude = longitude;
    }

    public String getDisplay_name() {
        return display_name;
    }

    public void setDisplay_name(String display_name) {
        this.display_name = display_name;
    }

    public String getPhoto_url() {
        return photo_url;
    }

    public void setPhoto_url(String photo_url) {
        this.photo_url = photo_url;
    }

    public String getReviewsID() {
        return reviewsID;
    }

    public void setReviewsID(String reviewsID) {
        this.reviewsID = reviewsID;
    }
}

File: app/src/main/java/com/kacper/krakgo/model/Place.kt
package com.kacper.krakgo.model

import android.os.Parcel
import android.os.Parcelable

/**
 * Created by kacper on 10/03/2018.
 */

class Place : Parcelable {
    var latitude: Double = -1.0
    var longitude: Double = -1.0
    lateinit var display_name: String
    lateinit var photo_url: String
    lateinit var reviewsID: String

    constructor() {
        //Constructor needed for firebase data parse
    }

    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(parcel: Parcel, i: Int) {
        parcel.writeDoubleArray(doubleArrayOf(latitude, longitude))
        parcel.writeStringArray(arrayOf(display_name, photo_url, reviewsID))
    }

    constructor(`in`: Parcel) {
        val strings = arrayOfNulls<String>(3)
        val doubles = DoubleArray(2)

        `in`.readDoubleArray(doubles)
        `in`.readStringArray(strings)

        this.display_name = strings[0]!!
        this.photo_url = strings[1]!!
        this.reviewsID = strings[2]!!

        this.latitude = doubles[0]
        this.longitude = doubles[1]
    }

    companion object {
        @JvmField val CREATOR = object : Parcelable.Creator<Place> {
            override fun createFromParcel(parcel: Parcel) = Place(parcel)

            override fun newArray(size: Int) = arrayOfNulls<Place>(size)
        }
    }
}
